packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -243,7 +243,7 @@ export function postMountWrapper(
     node.name = '';
   }
   node.defaultChecked = !node.defaultChecked;
-  node.defaultChecked = !node.defaultChecked;
+  node.defaultChecked = !!node._wrapperState.initialChecked;
```

Comment:
Just trying to think through all of the cases here:

**`checked` or `defaultChecked` are `true`**

```javascript
node.defaultChecked = true // node.defaultChecked = !node.defaultChecked
node.defaultChecked = true // node.defaultChecked = !!false
```

**`checked` or `defaultChecked` are `false`**

```javascript
node.defaultChecked = true  // node.defaultChecked = !node.defaultChecked
node.defaultChecked = false // node.defaultChecked = !!false
```

**`checked` and `defaultChecked` are `undefined`**

This can happen if you just declare a regular text input, ex: `<input value="test" />`

In this case, value detachment still needs to occur. An input might switch from `text` to `radio` later, and React doesn't fire the post mount hook behavior when that happens.

```javascript
node.defaultChecked = true  // node.defaultChecked = !node.defaultChecked
node.defaultChecked = false // node.defaultChecked = !!undefined
```

So I think this is good to go. Are there any other cases you can think of?

Reply:
Separately, part of me wonders if `node._wrapperState.initialChecked` should always be a boolean. Like we should do the `!!node._wrapperState.initialChecked` part [when initialChecked is assigned](https://github.com/dilidili/react/blob/146bb96425f1c350fa7299658c9a5b6b253b024d/packages/react-dom/src/client/ReactDOMFiberInput.js#L123-L124). 

However I don't think that's related to this issue, and I'm not sure what existing code relies on that value to be loosely null.

Reply:
Yeah. One thing at a time. I think this is a great follow-up, but I'd love to get this check attribute fix in first.